src/components/Main/MainView/MessageInput/DropdownSuggester.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div :class="$style.container" :style="style">
+    <div
+      :class="{ [$style.item]: true, [$style.first]: index === 0 }"
+      v-for="(candidate, index) in candidates"
+      :key="candidate"
+      @click="onClick(e, candidate)"
+    >
+      {{ candidate }}
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, SetupContext, computed, PropType } from 'vue'
+import Icon from '@/components/UI/Icon.vue'
+
+const useClickHandlers = (
+  context: SetupContext<{
+    select: (word: string) => string
+  }>
+) => {
+  const onClick = (_e: MouseEvent, word: string) => {
+    context.emit('select', word)
+  }
+  return { onClick }
+}
+
+// スタンプのプレビュー表示とか用(wordCompleterからexportしたほうが良さそう)
+// interface Candidate {
+//   text: string
+//   type:
+// }
+
+export default defineComponent({
+  name: 'DropdownSuggester',
+  // components: {
+  //   Icon
+  // },
+  props: {
+    // candidates: {
+    //   type: Array as PropType<Candidate[]>,
+    //   default: null
+    // }
+    candidates: {
+      type: Array as PropType<string[]>,
+      default: []
+    },
+    position: {
+      type: Object as PropType<{ top: number; left: number }>,
+      required: true
+    }
+  },
+  emits: {
+    select: (word: string) => word
```

Comment:
```suggestion
    select: (word: string) => true
```
返り値は引数が正しいものかだからここは常に`true`返せばいい(TypeScriptで保障されてるので)